subworkflows/local/utils_nfcore_detaxizer_pipeline/main.nf

```diff
@@ -0,0 +1,238 @@
+    UTILS_NFVALIDATION_PLUGIN (
+        help,
+        workflow_command,
+        pre_help_text,
+        post_help_text,
+        validate_params,
+        "nextflow_schema.json"
+    )
+
+    //
+    // Check config provided to the pipeline
+    //
+    UTILS_NFCORE_PIPELINE (
+        nextflow_cli_args
+    )
+    //
+    // Custom validation for pipeline parameters
+    //
+    validateInputParameters()
+
+    //
+    // Create channel from input file provided through params.input
+    //
+    Channel
+        .fromSamplesheet("input")
+        .set { ch_samplesheet }
+
+    emit:
+    samplesheet = ch_samplesheet
+    versions    = ch_versions
+}
+
+/*
+========================================================================================
+    SUBWORKFLOW FOR PIPELINE COMPLETION
+========================================================================================
+*/
+
+workflow PIPELINE_COMPLETION {
+
+    take:
+    email           //  string: email address
+    email_on_fail   //  string: email address sent on pipeline failure
+    plaintext_email // boolean: Send plain-text email instead of HTML
+    outdir          //    path: Path to output directory where results will be published
+    monochrome_logs // boolean: Disable ANSI colour codes in log output
+    hook_url        //  string: hook URL for notifications
+    multiqc_report  //  string: Path to MultiQC report
+
+    main:
+
+    summary_params = paramsSummaryMap(workflow, parameters_schema: "nextflow_schema.json")
+
+    //
+    // Completion email and summary
+    //
+    workflow.onComplete {
+        if (email || email_on_fail) {
+            completionEmail(summary_params, email, email_on_fail, plaintext_email, outdir, monochrome_logs, multiqc_report.toList())
+        }
+
+        completionSummary(monochrome_logs)
+
+        if (hook_url) {
+            imNotification(summary_params, hook_url)
+        }
+    }
+}
+
+/*
+========================================================================================
+    FUNCTIONS
+========================================================================================
+*/
+//
+// Check and validate pipeline parameters
+//
+def validateInputParameters() {
+    genomeExistsError()
+}
+
+//
+// Validate channels from input samplesheet
+//
+def validateInputSamplesheet(input) {
+    def (metas, fastqs) = input[1..2]
+
+    // Check that multiple runs of the same sample are of the same datatype i.e. single-end / paired-end
+    def endedness_ok = metas.collect{ it.single_end }.unique().size == 1
+    if (!endedness_ok) {
+        error("Please check input samplesheet -> Multiple runs of a sample must be of the same datatype i.e. single-end or paired-end: ${metas[0].id}")
+    }
+
+    return [ metas[0], fastqs ]
+}
+//
+// Get attribute from genome config file e.g. fasta
+//
+def getGenomeAttribute(attribute) {
```

Comment:
In Sarek and the Raredisease-pipeline (and some other pipeline), the function `getGenomeAttribute` was moved from `subworkflows/local/utils_nfcore....py` to `main.nf`:

https://github.com/maxulysse/nf-core_sarek/blob/f9d172321e109220425eca058bf5dea4016c824f/main.nf#L365-L372

@maxulysse and @ramprasadn will know why.